### ISSUE TITLE
JetBooking - Add half days for calender for check-in & check-out times. 

### ISSUE THREAD

**Issue by peterloran** [CrocoBlock/suggestions#5170]
I've seen a bookingsystem where they can put specific check-in and check-out times so a day can be booked half/half.

For example Family "A" books from monday to friday. (check-out time set: 11h for example).
Family "B" wants to book the same room from friday to sunday. (check-in date 13h for example). 

How to show it in the availability calender. Make the friday a triangle shaped day. This isn't a new feature to other plugins. But it would be nice to implement the same feature with JetBooking as well.

An example for the booking UX/UI: https://imgur.com/a/hGvyaiY
![example](https://user-images.githubusercontent.com/103453897/162811568-61bc9e04-e833-43ee-87f0-68fe89b5ba2f.jpg)
/
In this example, the family books on tuesday 18th of august at 14:00 for example, and checks out at 10:00 o clock in the morning at the 23rd of august on sunday.

**Comment by Floydtm** [comment]
We have an option Booking Period Maybe this will work for you https://prnt.sc/l8Wp-DTXQdCy

**Comment by peterloran** [comment]
No that doesn't really help. It's not the same thing. But thanks for the suggestion. Appreciate the thought.

**Comment by Clicktoweb** [comment]
> We have an option Booking Period Maybe this will work for you https://prnt.sc/l8Wp-DTXQdCy

With this option you enable the last day booked but not the first.. 
The suggestion of peterloran is essential for all hotel, B&B.. without this option there is always a free night between 2 stays

**Comment by pamanseau** [comment]
This feature is not only important for Hotels, B&B but also for many other but the CAR RENTAL template RENTGO from Crocoblock doesn't have the time of pickup and time of drop so as it is right now it's not usable for a car rental company.

**Comment by ahxzyn** [comment]
has this  ever been solved via an update? this is the main reason why I can't use jetbooking for a vacation rental site for a client

**Comment by nickj-git** [comment]
Yes, part day availability would be great. I've got a hall hire use case for example where a booking could be for two bookings on the same day, but no more than that. If someone places a morning or early afternoon booking, the space would still be available at night which currently the calendar would display as booked for that date.

**Comment by UraraReika** [comment]
Hi 👋

Thank you for the detailed explanation and the clear real-world example.

Unfortunately, we do not plan to introduce such functionality into our plugin. Now our team is working on more priority tasks. But in the future we do not exclude the possibility of adding this functionality to the core of our plugin.

For now I’ve created a small add-on plugin called JetBooking Half Day that demonstrates how this can be implemented on top of JetBooking without modifying core files - https://github.com/UraraReika/jet-booking-half-day. This solution keeps backward compatibility and does not interfere with the booking engine logic — it only improves calendar visualisation.

Thank you for your understanding.
